### ISSUE TITLE
Saving edits to a layer with JSON field throws an error

### ISSUE THREAD

**Issue by nextstopsun** [qgis/QGIS#30131]
Easy to reproduce:
1. Add a postgis layer with json (jsonb) column to a QGIS project.
2. Start editing and add a new record.
3. Click Save edits. 

**Boom!**
```
Could not commit changes to layer (your_layer_here)
Errors: ERROR: 1 feature(s) not added.

  Provider errors:
      PostGIS error while adding features: ERROR:  column "jsonb_column" is of type jsonb but expression is of type hstore
    LINE 1: ...::hstor...
                                                                 ^
    HINT:  You will need to rewrite or cast the expression.
```
Looks like QGIS casts jsonb to hstore for internal representation and forgets to cast back when inserting/updating postgresql table.

Found on QGIS 3.6

**Comment by elpaso** [comment]
I think jsonb is not yet supported.

**Comment by elpaso** [comment]
mmm looking at the code, it might be supported while reading but as you mentioned not when writing, I'll have a look at that.

**Comment by nextstopsun** [comment]
@elpaso As an offtopic: do you have any plans to add support for nested json/jsonb and json arrays?

**Comment by elpaso** [comment]
@nextstopsun my latest implementation in the library and in the provider does support all JSON variants (nested objects and arrays included), but I have no plans for adding GUI support for it.
See: https://github.com/qgis/QGIS/pull/30137/files#diff-99b101819133a316786604243f3abf6eR629

**Comment by nextstopsun** [comment]
@elpaso That's great. 
I'm ok with a text widget as long as it's supported. (Though formatting/syntax highlighting could be nice). 
Thanks a lot - waiting for it to be merged and land to 3.8.
